server/models/Classroom.coffee

```diff
@@ -44,13 +45,74 @@ ClassroomSchema.pre('save', (next) ->
     next()
 )
 
-
 ClassroomSchema.methods.isOwner = (userID) ->
   return userID.equals(@get('ownerID'))
 
 ClassroomSchema.methods.isMember = (userID) ->
   return _.any @get('members') or [], (memberID) -> userID.equals(memberID)
 
+ClassroomSchema.methods.generateCoursesData = co.wrap (isAdmin=false) ->
+  # Helper function for generating the latest version of courses
+  query = {}
+  query = {releasePhase: 'released'} unless isAdmin
+  courses = yield Course.find(query)
+  courses = Course.sortCourses courses
+  campaigns = yield Campaign.find({_id: {$in: (course.get('campaignID') for course in courses)}})
+  campaignMap = {}
+  campaignMap[campaign.id] = campaign for campaign in campaigns
```

Comment:
I misinterpreted this as `campaignMap[campaign.id] = campaign for campaign in campaigns` at first.
I'd rewrite it like this to make it more clear:
```
for campaign in campaigns
  campaignMap[campaign.id] = campaign
```